modules/lib/core/index.d.ts

```diff
@@ -52,19 +52,105 @@ export interface PublishInfo {
     first?: string;
 }
 
-export interface Component<Config extends object = object, Regions extends Record<string, Region> = Record<string, Region>> {
-    config: Config;
-    descriptor: string;
+// ─────────────────────────────────────────────────────────────────────────────
+// Component
+// ─────────────────────────────────────────────────────────────────────────────
+export interface FragmentComponent {
+    type: 'fragment'
+    fragment: string;
     path: string;
-    type: 'page' | 'layout' | 'part';
+}
+
+export interface LayoutComponent<
+    Descriptor extends string = string,
+    Config extends object = object,
+    Regions extends Record<string, LayoutRegion> = Record<string, LayoutRegion>
+> {
+    type: 'layout'
+    descriptor: Descriptor
+    config: Config
+    path?: string // Missing in fragmentPreview https://github.com/enonic/xp/issues/10116
+    regions: Regions;
+}
+
+export interface PartComponent<
+    Descriptor extends string = string,
```

Comment:
The thinking was that descriptor was the discriminating "id". So for a layout it could theoretically limit the shape of regions. But in the example above there is no "link" between descriptor and regions.